pkg/model/resource/utils.go

```diff
@@ -67,7 +65,29 @@ func APIPackagePathLegacy(repo, group, version string, multiGroup bool) string {
 	return path.Join(repo, "api", version)
 }
 
+// unpluralizedSuffixes is a list of resource suffixes that are the same plural and singular
+// This is only is only necessary because some bits of code are lazy and don't actually use the RESTMapper like they should.
+// TODO eliminate this so that different callers can correctly map to resources.  This probably means updating all
+// callers to use the RESTMapper they mean.
```

Comment:
I think we can just leave a reference as well, which could be enough.
(This should also resolve the lint error as well)

<img width="878" alt="image" src="https://github.com/kubernetes-sigs/kubebuilder/assets/18136486/2316ab00-2375-4888-8ee0-aa4684368ee6">